streaming/src/main/scala/org/apache/spark/streaming/dstream/PairDStreamFunctions.scala

```diff
@@ -350,6 +349,18 @@ class PairDStreamFunctions[K, V](self: DStream[(K, V)])
     )
   }
 
+  /** TODO: Add scala docs */
+  def trackStateByKey[S: ClassTag, T: ClassTag](
```

Comment:
Especially if there can be a doc that describes (K, V, S, T) and what their semantics are.